src/poly/commitment/verifier.rs

```diff
@@ -31,28 +31,26 @@ impl<C: CurveAffine> OpeningProof<C> {
             let u_y2 = u_x.square() * &u_x + &C::b();
             let u_y = u_y2.deterministic_sqrt();
             if u_y.is_none() {
-                return false;
+                return Err(Error::OpeningError);
             }
             let u_y = u_y.unwrap();
 
             C::from_xy(u_x, u_y).unwrap()
         };
 
-        let mut extra_scalars = Vec::with_capacity(self.rounds.len() * 2 + 4 + params.n as usize);
```

Comment:
We'll want to keep these working vectors around still, rather than trying to access the `MSM` and modify them. When we're ready later on in this function (after the challenge `c` is sampled), we'll just loop on `extra_scalars`/`extra_bases` and call `msm.add_term`.